frontend/src/components/ChoicesFilterSection.tsx

```diff
@@ -8,23 +8,48 @@ interface Props {
   title: string;
   value: string;
   choices: Record<string, string>;
+  exclusiveChoice: boolean;
   onChange: (value: string) => void;
 }
 
-const ChoicesFilterSection = ({ title, value, choices, onChange }: Props) => {
+const ChoicesFilterSection = ({
+  title,
+  value,
+  choices,
+  exclusiveChoice,
+  onChange,
+}: Props) => {
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (event.target.checked) {
-      onChange(event.target.name);
+    if (exclusiveChoice) {
+      // Only one value allowed
+      if (event.target.checked) {
+        onChange(event.target.name);
+      } else {
+        onChange('');
+      }
     } else {
-      onChange('');
+      // Multiple values allowed
+      let all_values = value ? value.split(',') : [];
+
+      if (event.target.checked) {
+        // Add the checked value
+        all_values.push(event.target.name);
+      } else {
+        // Remove the unchecked value
+        all_values = all_values.filter((item) => item !== event.target.name);
+      }
+      onChange(all_values.join(','));
     }
   };
 
   return (
     <TitledSection title={title}>
       <Box display="flex" flexDirection="column">
         {Object.entries(choices).map(([choiceValue, choiceLabel]) => {
-          const checked = value === choiceValue;
+          const checked = exclusiveChoice
+            ? value === choiceValue
+            : value != '' && value.split(',').includes(choiceValue);
```

Comment:
I could actually remove 'value != '' && '

Reply:
I'll fix it in the same commit as other incoming remarks

Reply:
The split would be computed once per object. Could you instead have sooner in the component `const valuesList = exclusiveChoice ? [value] : value.split(',')` and use this instead here?